client/src/app/site/motions/services/category-repository.service.ts

```diff
@@ -0,0 +1,89 @@
+import { Injectable } from '@angular/core';
+import { Category } from '../../../shared/models/motions/category';
+import { ViewCategory } from '../models/view-category';
+import { DataSendService } from '../../../core/services/data-send.service';
+import { Observable } from 'rxjs';
+import { DataStoreService } from '../../../core/services/data-store.service';
+import { BaseRepository } from '../../base/base-repository';
+
+/**
+ * Repository Services for Categorys
+ *
+ * The repository is meant to process domain objects (those found under
+ * shared/models), so components can display them and interact with them.
+ *
+ * Rather than manipulating models directly, the repository is meant to
+ * inform the {@link DataSendService} about changes which will send
+ * them to the Server.
+ */
+@Injectable({
+    providedIn: 'root'
+})
+export class CategoryRepositoryService extends BaseRepository<ViewCategory, Category> {
+    /**
+     * Creates a CategoryRepository
+     * Converts existing and incoming category to ViewCategories
+     * Handles CRUD using an observer to the DataStore
+     * @param DataSend
+     */
+    public constructor(protected DS: DataStoreService, private dataSend: DataSendService) {
+        super(DS, Category, [Category]);
+    }
+
+    protected createViewModel(category: Category): ViewCategory {
+        const id = category.id;
+        const name = category.name;
+        const prefix = category.prefix;
+        return new ViewCategory(category, id, prefix, name);
+    }
+
+    public create(update: any, viewCategory?: ViewCategory): Observable<any> {
+        const categories = this.DS.getAll(Category);
+        const categoryIds: number[] = [];
+        if (update instanceof Category) {
+            viewCategory = new ViewCategory(update);
+        }
+        if (update instanceof ViewCategory) {
+            viewCategory = update;
+        }
+        categories.forEach(category => {
+            categoryIds.push(category.id);
+        });
+        if (update.id in categoryIds) {
+            return this.update(update, viewCategory);
+        } else {
+            return this.dataSend.saveModel(viewCategory.category);
+        }
+    }
+
+    public update(update: any, viewCategory?: ViewCategory): Observable<any> {
+        let updateCategory: Category;
+        if (viewCategory) {
+            updateCategory = viewCategory.category;
+        } else {
+            updateCategory = new Category();
+        }
+        updateCategory.patchValues(update);
+        return this.dataSend.saveModel(updateCategory);
+    }
+
+    public delete(viewCategory: ViewCategory): Observable<any> {
+        const category = viewCategory.category;
+        return this.dataSend.delete(category);
+    }
+
+    /**
+     * Checks if a Catagory is on the server already
+     * @param category the category to check if it is already on the server
+     */
+    public isOnServer(category: ViewCategory): boolean {
```

Comment:
stuff in the repository, or even in the DataStore, is not necessarily on the server (caching).
Therefore, your function will not tell you if this is on the server or not.
What you could check is, if a category exists on the DataStore while `DS.`Getting-all the categories and comparing them with your `viewCat.category`.

- rename to `osInDataStore`
- use `DS.getAll(Categories)`